docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -153,6 +153,299 @@ $ quarkus-project-0.1-SNAPSHOT-runner -XX:PrintFlags=
+is the bcc/eBPF script that will be used to capture the `malloc` stacktraces with their allocated size.
+This script, and other typical bcc/eBPF scripts (e.g. `stackcount`), need to be given a process ID (PID).
+This makes a bit tricky when you want to trace startup,
+but you can use `gdb`
+(even if you haven't enabled debug info)
+to get around this obstacle because it allows you to stop the application at the first instruction.
+Let's start by running the native executable via `gdb`:
+
+[source,bash]
+----
+$ gdb --args ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Xmx128m
+...
+(No debugging symbols found in ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner)
+----
+
+`starti` is a `gdb` command that sets a temporary breakpoint at the very first instruction of the program's execution.
+
+[source,bash]
+----
+(gdb) starti
+Starting program: <..>/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Xmx128m
+Program stopped.
+0x00007ffff7fe4790 in _start () from /lib64/ld-linux-x86-64.so.2
+----
+
+Next invoke the bcc/eBPF script giving it the PID of the Quarkus process,
+so that it can track the `malloc` calls,
+capture stack traces and dump them to a file for post-processing:
+
+[source,bash]
+----
+$ sudo ./mallocstacks.py -p $(pidof code-with-quarkus-1.0.0-SNAPSHOT-runner) -f > out.stacks
+----
+
+Then go back to the `gdb` shell and instruct it to continue the startup procedure after hitting the first instruction:
+
+[source,bash]
+----
+(gdb) continue
+Continuing.
+[Thread debugging using libthread_db enabled]
+Using host libthread_db library "/lib64/libthread_db.so.1".
+[New Thread 0x7ffff65ff6c0 (LWP 3342)]
+...
+[New Thread 0x7fffc6ffd6c0 (LWP 3359)]
+__  ____  __  _____   ___  __ ____  ______
+ --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2023-02-09 18:02:32,794 INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT native (powered by Quarkus 2.16.1.Final) started in 0.011s. Listening on: http://0.0.0.0:8080
+2023-02-09 18:02:32,794 INFO  [io.quarkus] (main) Profile prod activated.
+2023-02-09 18:02:32,794 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
+----
+
+Once the startup has complete,
+press `Ctrl-C` on the `stackcount` shell.
+
+Then process the stacks file as a flamegraph.
+Note that the stacks generated by this script are already collapsed,
+so the flamegraph can be generated just like this:
+
+[source,bash]
+----
+$ cat out.stacks | ${FG_HOME}/flamegraph.pl --color=mem --title="malloc bytes Flame Graph" --countname="bytes" > out.svg
+----
+
+The flamegraph produced should look something like this:
+
+image::malloc-bytes.svg[malloc bytes flamegraph]
+
+This shows that most of the memory requested using `malloc` comes from epoll in Java NIO,
+but the overall amount allocated via `malloc` is barely 268KB.
```

Comment:
How do we know that it is barely 268KB?

Reply:
It's in the svg file. If you look at the very bottom line it'll tell you how much memory is allocated using `malloc` for all its calls:

![Screenshot 2023-02-13 at 11 50 41](https://user-images.githubusercontent.com/50187/218438609-6d9db0d4-a4af-4ee2-8f6c-f297c7ecb809.png)

Anything wrong with that? You can inspect the svg by opening the file in a different tab.

Reply:
No, nothing wrong with this. It's just not clear in the text where this comes from. I would add a hint :)

Reply:
I've just updated the explanation with a hint.